recipes/tsil/all/conanfile.py

```diff
@@ -0,0 +1,69 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.30.0"
+
+
+class TsilConan(ConanFile):
+    name = "tsil"
+    license = "GPL-2.0-or-later"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.niu.edu/spmartin/TSIL/"
+    description = "Two-loop Self-energy Integral Library"
+    topics = ("conan", "high-energy", "physics", "hep", "two-loop", "integrals")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "TSIL_SIZE": ["TSIL_SIZE_LONG", "TSIL_SIZE_DOUBLE"]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "TSIL_SIZE": "TSIL_SIZE_LONG"
+    }
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("TSIL does not support {}".format(self.settings.compiler))
```

Comment:
Did you try?

Reply:
Yes. :)
The reason why Visual Studio is not supported is that TSIL uses C99-conform complex arithmetic from `complex.h`. C99-conform complex arithmetic is not supported in Visual Studio until today, see https://docs.microsoft.com/en-us/cpp/c-runtime-library/complex-math-support?view=msvc-160

Reply:
I see, thanks for the link.